software/prepare_src/prepare_src.py

```diff
@@ -131,16 +139,151 @@ def prepare_file(src, tgt, is_hash):
         f.write('#include "../../../dll.h"\n')
         f.write(txt)
 
-def prepare_directory(src_dir_path, tgt_dir_path, is_hash):
+

```

Comment:
Could you specify which schemes require this form of `crypto_aead.h` or `crypto_hash.h`? All the algorithms I tried can be compiled even using an empty file for these headers and the lib can be utilized using a minimal function def header.

Reply:
So if the only modifications are updating the makefile and and adding the include directory with the with empty headers the following were the results for comparison .

The list of stuff that doesn't compile is still large.

with changes
rich@ubuntu:~/LWC/software/prepare_src (update_prepare_source)$ ll libs/crypto_aead/ | wc -l
873
without changes but including empty headers
rich@ubuntu:~/LWC/software/prepare_src (master)$ ll libs/crypto_aead/ | wc -l
803
rich@ubuntu:~/supercop-20200702$ ll crypto_aead/ | wc -l
479

with changes
rich@ubuntu:~/LWC/software/prepare_src (update_prepare_source)$ ll libs/crypto_hash/ | wc -l
209
without changes but including empty headers
rich@ubuntu:~/LWC/software/prepare_src (master)$ ll libs/crypto_hash/ | wc -l
99
rich@ubuntu:~/supercop-20200702$ ll crypto_hash/ | wc -l
161

Reply:
The following still are not supported due to directory format: (Could be supported)
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/tiger/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/lane512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/echosp256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/nasha512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/cheetah256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/luffa256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/sarmal256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/shavite3256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/shavite3512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/sarmal512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/fugue256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/edonr256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/bblake512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/cheetah512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/luffa384/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/keccakc256treed2/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/luffa512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/essence512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/lux256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/edonr512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/groestl512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/echo512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/fugue384/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/skein256256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/bblake256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/essence256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/k12/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/essence224/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/lane256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/essence384/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/hamsi512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/echosp512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/fugue2/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/nasha256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/fugue512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/blake3/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/echo256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/keccakc512treed2/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/skein10241024/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/lux512/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/shabal256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/hamsi/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_hash/groestl256/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_aead/romulusn1plusv13/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_aead/skinnyaeadtk3128128plusv1/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:/home/rich/supercop-20200702/crypto_aead/romulusm1plusv13/ has no subdirectories
	 ref, simple or openssl
WARNING:__main__:crypto_aead/polawisv1/ref/crypto_aead.h will not build successfully!

Reply:
Even with these changes the error log is still 600 lines long.